mlrun/datastore/targets.py

```diff
@@ -1467,6 +1472,469 @@ def as_df(
+        )
+        table = self._resource.uri
+        graph.add_step(
+            name=self.name or "MongoDBTarget",
+            after=after,
+            graph_shape="cylinder",
+            class_name="storey.NoSqlTarget",
+            columns=column_list,
+            header=True,
+            table=table,
+            index_cols=key_columns,
+            # storage_options=self._get_store().get_storage_options(),
+            **self.attributes,
+        )
+
+    def as_df(
+        self,
+        columns=None,
+        df_module=None,
+        entities=None,
+        start_time=None,
+        end_time=None,
+        time_column=None,
+        **kwargs,
+    ):
+        query = {}
+        if time_column:
+            # creat time query and adding it to the existed query.
+            time_query = {time_column: {}}
+            if start_time:
+                time_query[time_column]["$gte"] = start_time
+            if end_time:
+                time_query[time_column]["$lt"] = end_time
+            if time_query[time_column] and query:
+                query.update(time_query)
+            elif time_query[time_column] and not query:
+                query = time_query
+
+        from pymongo import MongoClient
+
+        connection_string, db_name, collection_string = self._parse_url()
+        mongodb_client = MongoClient(connection_string)
+        db = mongodb_client[db_name]
+        collection = db[collection_string]
+        cursor = collection.find(query)
+
+        df = pd.DataFrame(list(cursor))
+        if "_id" in df.columns:
+            df["_id"] = str(df["_id"])
+        mongodb_client.close()
+        return df
+
+    def write_dataframe(
+        self, df, key_column=None, timestamp_key=None, chunk_id=0, **kwargs
+    ):
+        if hasattr(df, "rdd"):
+            raise ValueError("Spark is not supported")
+        else:
+            from pymongo import MongoClient
+
+            connection_string, db_name, collection_string = self._parse_url()
+            mongodb_client = MongoClient(connection_string)
+            db = mongodb_client[db_name]
+            collection = db[collection_string]
+
+            data = df.to_dict(orient="records")
+            collection.insert_many(data)
+            mongodb_client.close()
+
+    def _parse_url(self):
+        path = self.path[len("mdb:///") :]
+        return path.split("///")
+
+    def purge(self):
+        pass
+
+
+class SqlDBTarget(BaseStoreTarget):
+    kind = TargetTypes.sql
+    is_online = True
+    support_spark = False
+    support_storey = True
+    _SQL_DB_PATH_STRING_ENV_VAR = "SQL_DB_PATH_STRING"
+
+    def __init__(
+        self,
+        name: str = "",
+        path=None,
+        attributes: typing.Dict[str, str] = None,
+        after_step=None,
+        partitioned: bool = False,
+        key_bucketing_number: typing.Optional[int] = None,
+        partition_cols: typing.Optional[typing.List[str]] = None,
```

Comment:
in my opinion, it's much more readable to: `from typing import List, Optional, Dict, ...` than wrrite `typing.X` everytime. Also, if your parameterr has a default value of `x = None` the `Optional` is redundant. `Optional` is used for `dataclass`es.